source/config.go

```diff
@@ -14,44 +14,6 @@
 
 package source
```

Comment:
maybe we should rename the file? it doesn't have any config info anymore

Reply:
How would you name it? 🙃

Reply:
hmmm `modes`? `modeTypes`? maybe even move the types to another file?
just because we usually have the config struct under this name, and it's usually the first file I check, so it's a bit confusing

Reply:
I couldn't find a good name, it got me thinking something is fishy. I realized that we can move `SourceConfig` into `source` and have a `source.Config` instead, then it also made sense to have these types in the `config.go` file as they are types related to fields in the config. I think it's cleaner now 🙂 I made the same changes for the destination config.